src/super_gradients/training/samples/depth_estimation_sample.py

```diff
@@ -0,0 +1,28 @@
+import dataclasses
+from typing import Union
+
+import numpy as np
+import torch
+from PIL.Image import Image
+
+__all__ = ["DepthEstimationSample"]
+
+
+@dataclasses.dataclass
+class DepthEstimationSample:
+    """
+    A dataclass representing a single depth estimation sample.
+    Contains input image and depth map.
+
+    :param image:              Image of [H, W, (C if colorful)] shape.
+    :param depth_map:          Depth map of [H, W] shape.
+    """
+
+    __slots__ = ["image", "depth_map"]
+
+    image: Union[np.ndarray, torch.Tensor]
+    depth_map: Union[np.ndarray, torch.Tensor]
+
+    def __init__(self, image: Union[np.ndarray, torch.Tensor, Image], depth_map: Union[np.ndarray, torch.Tensor, Image]):
```

Comment:
No need for torch.Tensor and especially for PIL.Image support here. In OD it was more for backward compatibility needs. Whenever possible let's stick to `np.ndarray` in the sample class.
The only place where we need to go to torch - is the collate function.